docs/source/messenger.rst

```diff
@@ -0,0 +1,113 @@
+  LICENSE file in the root directory of this source tree. An additional grant
+  of patent rights can be found in the PATENTS file in the same directory.
+
+Using Facebook Messenger
+========================
+**Authors**: Jack Urbanek
+
+In ParlAI, you can allow people on Facebook to participate in a ParlAI world as an agent.
+
+Facebookers can be viewed as another type of agent in ParlAI, and hence person-to-person, person-to-bot, or multiple people and bots in group chat can all talk to each other within the same framework.
+
+Facebook agents communicate in observation/action dict format, the same as all other agents in ParlAI. During the conversation, messages are delivered to a person's inbox from the page that you have set up for ParlAI.
+
+.. figure:: _static/img/mturk-small.png
+   :align: center
+
+   *Example: TODO put image of a chat window here*
+
+Each messenger task has at least one messenger agent that connects to ParlAI using the Facebook messenger Send/Receive API, encapsulated as a ``MessengerAgent`` object.
+
+Each messenger task also consists of a ``World`` where all agents live and interact within.
+
+Messenger tasks can be grouped together within an ``Overworld`` which can spawn the subtasks and allow people to pick between multiple conversations.
+
+Example Tasks
+-------------
+
+We provide two examples of using Facebook Messenger with ParlAI:
+
+- `QA Data Collection <https://github.com/facebookresearch/ParlAI/blob/master/parlai/messenger/tasks/qa_data_collection/>`__: collect questions and answers from people, given a random Wikipedia paragraph from SQuAD.
+- `Overworld Demo <https://github.com/facebookresearch/ParlAI/blob/master/parlai/mturk/tasks/overworld_demo/>`__: let people select between three different subtasks, namely an echo bot, a demo of onboarding data collection, and a random chat.
+
+Task 1: Collecting Data
+^^^^^^^^^^^^^^^^^^^^^^^
+
+One of the biggest use cases of Messenger in ParlAI is to connect people and collect natural language data.
+
+As an example, the `QA Data Collection task <https://github.com/facebookresearch/ParlAI/blob/master/parlai/messenger/tasks/qa_data_collection/>`__ does the following:
+
+1. Pick a random Wikipedia paragraph from SQuAD dataset.
+2. Ask a person to provide a question given the paragraph.
+3. Ask the person to provide an answer to their question.
+
+In ``QADataCollectionWorld``, there are two agents: one is the person (``MessengerAgent``), the other is the task agent (``DefaultTeacher`` from SQuAD) that provides the Wikipedia paragraph.
+
+The ``QADataCollectionWorld`` uses ``turn_index`` to denote what stage the conversation is at. One *turn* means that ``world.parley()`` has been called once.
+
+After two turns, the task is finished, and the person's work can be saved during the ``World.shutdown()`` call.
+
+
+Task 2: Exposing People to Multiple Tasks
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TODO: Describe this.
+
+
+Creating Your Own Task
+----------------------
```

Comment:
ha maybe worth cutting all the mturk stuff out for now--can be pasted back in from the other doc when you have time

Reply:
Updating it right now!